service/history/queue/transfer_queue_processor_base.go

```diff
@@ -173,8 +179,12 @@ func (t *transferQueueProcessorBase) Stop() {
 	t.logger.Info("Transfer queue processor state changed", tag.LifeCycleStopping)
 	defer t.logger.Info("Transfer queue processor state changed", tag.LifeCycleStopped)
 
-	t.nextPollTimer.Close()
 	close(t.shutdownCh)
+	t.processingLock.Lock()
+	for _, timer := range t.backoffTimer {
+		timer.Stop()
+	}
```

Comment:
Do you need to change shouldProcess to false?

Reply:
We don't really care about the value in `shouldProcess` actually. As long as the `shutdownCh` is closed the processing loop will be stopped. But I can set it to false, so that the for loop in `processQueueCollections` can stop early. Will update.

Reply:
Updated.